### PR TITLE
Add support for finding federated plugins

### PR DESCRIPTION
**Pull request by hterkelsen**

### CODE REVIEW

File: test/list_command_test.dart [hunk 1]
```diff
@@ -102,5 +103,37 @@ void main() {
 
       cleanupPackages();
     });
+
+    test('lists plugins using multi-plugin layout', () async {
```
Comment (amirh) [inline test/list_command_test.dart]: nit: s/multi-plugin/federated plugin/

File: test/list_command_test.dart [hunk 2]
```diff
@@ -102,5 +103,37 @@ void main() {
...
+        unorderedEquals(<String>[
+          '/packages/plugin1',
+          '/packages/my_plugin/my_plugin',
+          '/packages/my_plugin/my_plugin_web',
+          '/packages/my_plugin/my_plugin_macos',
+        ]),
+      );
+
+      cleanupPackages();
+    });
```
Comment (amirh) [inline test/list_command_test.dart]: nit: we should also test with the plugins argument set.

File: test/list_command_test.dart (hunk 1 above)
Comment (hterkelsen) [reply test/list_command_test.dart]: Done.

File: test/list_command_test.dart (hunk 2 above)
Comment (hterkelsen) [reply test/list_command_test.dart]: Done.

Comment (amirh) [reply test/list_command_test.dart]: I meant `--plugins plugin1` (as this affects the tested logic)

Comment (hterkelsen) [reply test/list_command_test.dart]: I will add a test in a follow-up PR. Do we want `--plugins=my_plugin` to mean that `/packages/my_plugin/my_plugin` and `/packages/my_plugin/my_plugin_web` both match, or that neither of them match?

Comment (amirh) [reply test/list_command_test.dart]: Well now that we're thinking about it, I think ideally:

`--plugins=my_plugin/<package>` - will match just the specified subfolder
`--plugins=my_plugin` - will match all packages under `my_plugin`

**Pull request merged**